## UN Comtrade에서 무역 데이터 불러오기

#### param
- typeCode = "C": 상품 / "S": 서비스
- freqCode = "A": 연간 데이터 / "M": 월별 데이터
- period = "2022": 연간 / "202205": 월, (여려 연도나 월일 경우 $\Rightarrow$ "2021,2022,2023")
- reporterCode: 기준이 되는 나라의 코드
- partnerCode: 상대 나라의 코드
- clCode = "HS" (품목 분류 기준, 건들 필요 X)
- cmdCode = "72": 철강 / "TOTAL": 전체 품목... (품목 코드)
- flowCode = "X": 수출 / "M": 수입
- partner2Code: 제2 중간국
- customsCode: 통관방식
- motCode: 운송수단

In [2]:
import pandas as pd
import comtradeapicall
from src.config import get_comtrade_api_key

In [ ]:
API_KEY = get_comtrade_api_key()

In [4]:
df = comtradeapicall.getFinalData(
    API_KEY,
    typeCode='C',                     # 상품
    freqCode='A',                     # 연간
    clCode='HS',                      # HS 분류
    period='2020,2021,2022,2023,2024',
    reporterCode='704',               # 베트남(기준)
    cmdCode='72',                     # 철강
    flowCode='X',                     # 수출
    partnerCode='410',                # 한국 => [베트남 -> 한국]
    partner2Code=None,
    customsCode=None,
    motCode=None,
    maxRecords=2500,
    format_output='JSON',
    aggregateBy=None,
    breakdownMode='classic',
    countOnly=None,
    includeDesc=True
)

In [5]:
df

,typeCode,freqCode,refPeriodId,refYear,refMonth,period,reporterCode,reporterISO,reporterDesc,flowCode,...,netWgt,isNetWgtEstimated,grossWgt,isGrossWgtEstimated,cifvalue,fobvalue,primaryValue,legacyEstimationFlag,isReported,isAggregate
0,C,A,20200101,2020,52,2020,704,VNM,Viet Nam,X,...,0.0,True,0.0,False,0.0,1.764312e+08,1.764312e+08,4,False,True
1,C,A,20210101,2021,52,2021,704,VNM,Viet Nam,X,...,0.0,True,0.0,False,0.0,3.566538e+08,3.566538e+08,4,False,True
2,C,A,20220101,2022,52,2022,704,VNM,Viet Nam,X,...,0.0,True,0.0,False,NaN,5.804562e+08,5.804562e+08,4,False,True
3,C,A,20230101,2023,52,2023,704,VNM,Viet Nam,X,...,0.0,True,0.0,False,NaN,4.640595e+08,4.640595e+08,4,False,True


In [6]:
df.iloc[1]

,1
typeCode,C
freqCode,A
refPeriodId,20210101
refYear,2021
refMonth,52
period,2021
reporterCode,704
reporterISO,VNM
reporterDesc,Viet Nam
flowCode,X


---
### UN Comtrade 컬럼

| 컬럼명                      | 값 예시           | 의미          | 해석                    |
| ------------------------ | -------------- | ----------- | --------------------- |
| typeCode                 | C              | 데이터 유형      | 상품(Commodity)         |
| freqCode                 | A              | 빈도          | 연간 데이터                |
| refPeriodId              | 20210101       | 기간 ID       | 기준 날짜 (2021-01-01 형태) |
| refYear                  | 2021           | 연도          | 해당 데이터 연도             |
| refMonth                 | 52             | 월           | 연간 데이터에서는 의미 없음 (내부값) |
| period                   | 2021           | 기간          | 연도 기준                 |
| reporterCode             | 704            | 보고국 코드      | 베트남                   |
| reporterISO              | VNM            | ISO 코드      | 베트남                   |
| reporterDesc             | Viet Nam       | 국가명         | 베트남                   |
| flowCode                 | X              | 흐름 코드       | 수출                    |
| flowDesc                 | Export         | 흐름          | 수출                    |
| partnerCode              | 410            | 상대국 코드      | 한국                    |
| partnerISO               | KOR            | ISO 코드      | 한국                    |
| partnerDesc              | Rep. of Korea  | 상대국명        | 대한민국                  |
| partner2Code             | 0              | 제2국 코드      | 없음                    |
| partner2ISO              | W00            | 제2국 ISO     | 전체(세계)                |
| partner2Desc             | World          | 제2국명        | 세계                    |
| classificationCode       | H5             | 분류 버전       | HS 2017 (H5)          |
| classificationSearchCode | HS             | 분류 기준       | HS 코드                 |
| isOriginalClassification | True           | 원분류 여부      | HS 기준 그대로             |
| cmdCode                  | 72             | 품목 코드       | 철강                    |
| cmdDesc                  | Iron and steel | 품목명         | 철강                    |
| aggrLevel                | 2              | 집계 수준       | 대분류 수준 (2단위)          |
| isLeaf                   | False          | 말단 여부       | 최하위 아님                |
| customsCode              | C00            | 통관 코드       | 전체 통관 기준              |
| customsDesc              | TOTAL CPC      | 통관 설명       | 전체 통관                 |
| mosCode                  | 0              | 거래 방식       | 전체                    |
| motCode                  | 0              | 운송수단 코드     | 전체                    |
| motDesc                  | TOTAL MOT      | 운송수단        | 전체                    |
| qtyUnitCode              | -1             | 수량 단위 코드    | 없음                    |
| qtyUnitAbbr              | N/A            | 수량 단위       | 없음                    |
| qty                      | 0.0            | 수량          | 없음                    |
| isQtyEstimated           | False          | 수량 추정 여부    | 실제값                   |
| altQtyUnitCode           | -1             | 대체 단위 코드    | 없음                    |
| altQtyUnitAbbr           | N/A            | 대체 단위       | 없음                    |
| altQty                   | 0.0            | 대체 수량       | 없음                    |
| isAltQtyEstimated        | False          | 대체 수량 추정 여부 | 실제값                   |
| netWgt                   | 0.0            | 순중량         | 없음 (집계 데이터라 비어있음)     |
| isNetWgtEstimated        | True           | 순중량 추정 여부   | 추정값                   |
| grossWgt                 | 0.0            | 총중량         | 없음                    |
| isGrossWgtEstimated      | False          | 총중량 추정 여부   | 실제값                   |
| cifvalue                 | 0.0            | CIF 가격      | 수입값이라 없음              |
| fobvalue                 | 356,653,803    | FOB 가격      | 수출 금액                 |
| primaryValue             | 356,653,803    | 주요 금액     | 실제 분석용 금액             |
| legacyEstimationFlag     | 4              | 추정 플래그      | 일부 추정 포함              |
| isReported               | False          | 보고 여부       | 파트너 기준 데이터            |
| isAggregate              | True           | 집계 여부       | 합계 데이터                |

---